SleepEngine/Engine/Components/HierarchyNode.cpp

```diff
@@ -27,26 +43,32 @@ Object* HierarchyNode::addChild(std::unique_ptr<Object>&& child)
 
     childNode->setParent(m_object);
 
-    m_children.emplace_back(std::move(child));
-    return childHandle;
+    m_children.emplace_back(child);
+    return child;
 }
 
-void HierarchyNode::removeChild(Object* child)
+void HierarchyNode::removeChild(Object const* child)
 {
-    auto const isChildToDelete = [child] (std::unique_ptr<Object> const& element)
+    auto const [found, it] = find(m_children, child);
+
+    if (!found)
     {
-        return element.get() == child;
-    };
+        LOG_AND_FAIL("'removeChild' called for object that is not our child");
+        return;
+    }
 
-    auto const childToDelete = findIf(m_children, isChildToDelete);
+    m_children.erase(it);

```

Comment:
Erase but no remove?

Reply:
Yes. 
When you want to completely remove the object you use `Object::removeLater()`, this is not the same thing.
This `removeChild` is being used when:
- detaching from the parent;
- changing parent to other one

Reply:
but erase + remove  would delete only pointer from m_children. As it is, nothing is removed, yes?

Reply:
Not from children. Please, look at `Object::removeLater` implementation